### PR TITLE
Solve flakiness in kb3 cluster

### PR DESCRIPTION
**Pull request by ybbbby**
Remove resource quotes restriction.
Grant controller with events related operations.
Resolve a cherry-pick bug left in https://github.com/grpc/test-infra/pull/203

### CODE REVIEW

File: controllers/loadtest_controller.go [hunk 1]
```diff
@@ -270,12 +271,12 @@ func (r *LoadTestReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		adjustAvailabilityForDefaults := func(defaultPoolKey, defaultPoolName string) bool {
 			if c, ok := missingPods.NodeCountByPool[defaultPoolKey]; ok && c > 0 {
 				if defaultPoolName == "" {
-					log.Error(errNonexistentPool, "default pool is not defined or does not existed in the cluster", "requestedDefaultPool", defaultPoolKey)
+					logger.Error(errNonexistentPool, "default pool is not defined or does not existed in the cluster", "requestedDefaultPool", defaultPoolKey)
```
Comment (wanlin31) [inline controllers/loadtest_controller.go:274]: Wondering how would it look like before.

File: controllers/loadtest_controller.go [hunk 2]
```diff
@@ -58,6 +58,7 @@ type LoadTestReconciler struct {
 // +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="",resources=nodes,verbs=get;list;watch
 // +kubebuilder:rbac:groups="",resources=nodes/status,verbs=get
+// +kubebuilder:rbac:groups="",resources=events,verbs=get;list;watch;create;update;patch;delete
```
Comment (wanlin31) [inline controllers/loadtest_controller.go:61]: Do we need all of the verbs? `get;list;watch;create;update;patch;delete`

File: controllers/loadtest_controller.go (hunk 1 above)
Comment (ybbbby) [reply controllers/loadtest_controller.go:274]: I did modify locally, but failed to commit in my last pr. It would give build error.

**Review by wanlin31 — APPROVED** [review]
LGTM just have some questions

File: controllers/loadtest_controller.go (hunk 2 above)
Comment (ybbbby) [reply controllers/loadtest_controller.go:61]: Good point, only `create` is observed when controller crashed and restarted. 
I assume we can grant controller all the power to deal with events, no?

**Review by paulosjca — APPROVED** [review]

**Pull request merged**